astropy/nddata/ccddata.py

```diff
@@ -0,0 +1,568 @@
+        Metadata for this object. "Metadata" here means all information that
+        is included with this object but not part of any other attribute
+        of this particular object, e.g. creation date, unique identifier,
+        simulation parameters, exposure time, telescope name, etc.
+
+    unit : `~astropy.units.Unit` or str, optional
+        The units of the data.
+        Default is ``None``.
+
+        .. warning::
+
+            If the unit is ``None`` or not otherwise specified it will raise a
+            ``ValueError``
+
+    Raises
+    ------
+    ValueError
+        If the ``uncertainty`` or ``mask`` inputs cannot be broadcast (e.g.,
+        match shape) onto ``data``.
+
+    Methods
+    -------
+    read(\\*args, \\**kwargs)
+        ``Classmethod`` to create an CCDData instance based on a ``FITS`` file.
+        This method uses :func:`fits_ccddata_reader` with the provided
+        parameters.
+    write(\\*args, \\**kwargs)
+        Writes the contents of the CCDData instance into a new ``FITS`` file.
+        This method uses :func:`fits_ccddata_writer` with the provided
+        parameters.
+
+    Notes
+    -----
+    `~astropy.nddata.CCDData` objects can be easily converted to a regular
+     Numpy array using `numpy.asarray`.
+
+    For example::
+
+        >>> from astropy.nddata import CCDData
+        >>> import numpy as np
+        >>> x = CCDData([1,2,3], unit='adu')
+        >>> np.asarray(x)
+        array([1, 2, 3])
+
+    This is useful, for example, when plotting a 2D image using
+    matplotlib.
+
+        >>> from astropy.nddata import CCDData
+        >>> from matplotlib import pyplot as plt   # doctest: +SKIP
+        >>> x = CCDData([[1,2,3], [4,5,6]], unit='adu')
+        >>> plt.imshow(x)   # doctest: +SKIP
+
+    """
+    def __init__(self, *args, **kwd):
+        if 'meta' not in kwd:
+            kwd['meta'] = kwd.pop('header', None)
+        if 'header' in kwd:
+            raise ValueError("can't have both header and meta.")
+
+        super(CCDData, self).__init__(*args, **kwd)
+
+        # Check if a unit is set. This can be temporarly disabled by the
+        # _CCDDataUnit contextmanager.
+        if _config_ccd_requires_unit and self.unit is None:
+            raise ValueError("a unit for CCDData must be specified.")
+
+    @property
+    def data(self):
+        return self._data
+
+    @data.setter
+    def data(self, value):
+        self._data = value
+
+    @property
+    def wcs(self):
+        return self._wcs
+
+    @wcs.setter
+    def wcs(self, value):
+        self._wcs = value
+
+    @property
+    def unit(self):
+        return self._unit
+
+    @unit.setter
+    def unit(self, value):
+        self._unit = u.Unit(value)
+
+    @property
+    def header(self):
+        return self._meta
+
+    @header.setter
+    def header(self, value):
+        self.meta = value
+
+    @property
+    def meta(self):
+        return self._meta
+
+    @meta.setter
```

Comment:
I don't think `meta` needs to be overridden. If I remember correctly that was some compatibility code because I only changed `meta` to the [`MetaData`](https://github.com/astropy/astropy/blob/master/astropy/utils/metadata.py#L365) descriptor in 1.2.

But it will behave slightly differently so you can also leave it in.